src/intervalmap.rs

```diff
@@ -0,0 +1,1524 @@
+        !self.node_ref(node_idx, Node::is_sentinel)
+    }
+
+    /// Find all intervals in the map that overlaps with the given interval.
+    ///
+    /// # Example
+    /// ```rust
+    /// use interval_map::{Interval, IntervalMap};
+    ///
+    /// let mut map = IntervalMap::new();
+    /// map.insert(Interval::new(1, 3), ());
+    /// map.insert(Interval::new(2, 4), ());
+    /// map.insert(Interval::new(6, 7), ());
+    /// map.insert(Interval::new(7, 11), ());
+    /// assert_eq!(map.find_all_overlap(&Interval::new(2, 7)).len(), 3);
+    /// map.remove(&Interval::new(1, 3));
+    /// assert_eq!(map.find_all_overlap(&Interval::new(2, 7)).len(), 2);
+    /// ```
+    #[inline]
+    pub fn find_all_overlap(&self, interval: &Interval<T>) -> Vec<(&Interval<T>, &V)> {
+        if self.node_ref(self.root, Node::is_sentinel) {
+            Vec::new()
+        } else {
+            self.find_all_overlap_inner_unordered(self.root, interval)
+        }
+    }
+
+    /// Return reference to the value corresponding to the key.
+    ///
+    /// # Example
+    /// ```rust
+    /// use interval_map::{Interval, IntervalMap};
+    ///
+    /// let mut map = IntervalMap::new();
+    /// map.insert(Interval::new(1, 3), 1);
+    /// map.insert(Interval::new(7, 11), 4);
+    /// assert_eq!(map.get(&Interval::new(1, 3)), Some(&1));
+    /// assert_eq!(map.get(&Interval::new(7, 11)), Some(&4));
+    /// assert_eq!(map.get(&Interval::new(5, 17)), None);
+    /// ```
+    #[inline]
+    pub fn get(&self, interval: &Interval<T>) -> Option<&V> {
+        self.search_exact(interval)
+            .map(|idx| self.node_ref(idx, Node::value))
+    }
+
+    /// Return a reference to the value corresponding to the key.
+    ///
+    /// # Example
+    /// ```rust
+    /// use interval_map::{Interval, IntervalMap};
+    ///
+    /// let mut map = IntervalMap::new();
+    /// map.insert(Interval::new(3, 5), 0);
+    /// map.get_mut(&Interval::new(3, 5)).map(|v| *v += 1);
+    /// assert_eq!(map.get(&Interval::new(3, 5)), Some(&1));
+    /// ```
+    #[inline]
+    pub fn get_mut(&mut self, interval: &Interval<T>) -> Option<&mut V> {
+        self.search_exact(interval)
+            .map(|idx| self.node_mut(idx, Node::value_mut))
+    }
+
+    /// Get an iterator over the entries of the map, sorted by key.
+    #[inline]
+    #[must_use]
+    pub fn iter(&self) -> Iter<'_, T, V, Ix> {
+        Iter::new(self)
+    }
+
+    /// Get an iterator over the entries of the map, sorted by key.
+    #[inline]
+    #[must_use]
+    pub fn into_iter(self) -> IntoIter<T, V, Ix> {
+        IntoIter::new(self)
+    }
+
+    #[inline]
+    pub fn unsorted_iter(&self) -> UnsortedIter<T, V, Ix> {
+        UnsortedIter::new(self)
+    }
+
+    #[inline]
+    pub fn filter_iter<'a, 'b: 'a>(&'a self, query: &'b Interval<T>) -> FilterIter<T, V, Ix> {
+        FilterIter::new(self, query)
+    }
+
+    #[inline]
+    pub fn contain(&self, interval: &Interval<T>) -> bool {
```

Comment:
Should be `contains`.

Reply:
I noticed that we used `insert`, `remove`, `overlap` etc, so to keep formal consistency, I used `contain` not `contains`.

Reply:
Do you have any suggestions?

Reply:
It should be "contains", because:
"The map contains an element."
"Insert an element into the map."

Reply:
I agree with you, so `overlap` should also be `overlaps`? Because it's "Check if an interval in the map `overlaps` with the given interval"

Reply:
> I agree with you, so `overlap` should also be `overlaps`? Because it's "Check if an interval in the map `overlaps` with the given interval"

Yeah I agree.